collect_app/src/main/java/org/odk/collect/android/injection/config/AppDependencyModule.java

```diff
@@ -265,4 +271,14 @@ public BackgroundWorkManager providesBackgroundWorkManager() {
     public NetworkStateProvider providesConnectivityProvider() {
         return new ConnectivityProvider();
     }
+
+    @Provides
+    public QRCodeGenerator providesQRCodeGenerator() {
+        return new QRCodeGenerator() {
+            @Override
+            public Observable<Bitmap> generateQRCode(Collection<String> selectedPasswordKeys) {
+                return QRCodeUtils.getQRCodeGeneratorObservable(selectedPasswordKeys);
```

Comment:
If possible it'd be nicer for `QRCodeUtils.getQRCodeGeneratorObservable`'s logic to move to this implementation (and probably have it in its own class file). Wrapping like this is a nice trick for dealing with statics we don't control but we do control this one!

Reply:
Is it okay to break up QRCodeUtils logic like this? Because it uses certain functions/static constants that aren't accessible outside the `QRCodeUtils` or outside the `utils` package? 

My concern is that If this is the only thing that needs to be moved out, I'm not sure if it's worth it to separate it from the other QRCodeUtils functions which may decrease readability.

Reply:
I do see your point but it seems to be that everything in that utils provides information about a single QR code image (the image itself, it's path etc) other than `decodeFromBitmap`. Could it be broken into a `QRCodeGenerator` and a `QRCodeReader` interface?

Reply:
If its okay with you, I went ahead and broke it into `QRCodeGenerator` interface, and left the remaining functions untouched in the `QRCodeUtils.java`